pkg/storage/bloom/v1/schema.go

```diff
@@ -0,0 +1,130 @@
+package v1
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/pkg/errors"
+
+	"github.com/grafana/loki/v3/pkg/chunkenc"
+	"github.com/grafana/loki/v3/pkg/util/encoding"
+)
+
+type Version byte
+
+func (v Version) String() string {
+	return fmt.Sprintf("v%d", v)
+}
+
+const (
+	magicNumber = uint32(0xCA7CAFE5)
+
+	// Add new versions below
+	V1 Version = iota
+	// V2 supports single series blooms encoded over multiple pages
+	// to accommodate larger single series
+	V2
+	// V2 indicated schema for indexed structured metadata
+	V3
+
+	CurrentSchemaVersion = V3
+)
+
+var (
+	SupportedVersions = []Version{V3}
+
+	ErrInvalidSchemaVersion     = errors.New("invalid schema version")
+	ErrUnsupportedSchemaVersion = errors.New("unsupported schema version")
+)
+
+type Schema struct {
+	version                Version
+	encoding               chunkenc.Encoding
+	nGramLength, nGramSkip uint64
```

Comment:
IIUC, we are going to remove this once we remove the ngrams, right? But we will keep V3